ConfiguredClientsStrategy.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Http\HttplugBundle;
+
+use Http\Client\HttpClient;
+use Http\Discovery\Exception\StrategyUnavailableException;
+use Http\Discovery\HttpClientDiscovery;
+use Http\Discovery\Strategy\DiscoveryStrategy;
+
+/**
+ * A strategy that provide clients configured with HTTPlug bundle.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class ConfiguredClientsStrategy implements DiscoveryStrategy
+{
+    /**
+     * @var HttpClient
+     */
+    private static $client;
+
+    /**
+     * @param HttpClient $httpClient
+     */
+    public function __construct(HttpClient $httpClient)
+    {
+        static::$client = $httpClient;
+
+        HttpClientDiscovery::prependStrategy(self::class);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getCandidates($type)
+    {
+        if (static::$client !== null && $type == 'Http\Client\HttpClient') {
```

Comment:
We could, We don't do it in [CommonClassesStrategy](https://github.com/php-http/discovery/blob/v0.9.1/src/Strategy/CommonClassesStrategy.php). But I don't know why i didn't use the ::class constant there. 

I should probably update here anyhow. Thank you.